### DISCUSSION TITLE
Triangle mesh with more than 130k triangles seems to get broken / squished along the X axis

### DISCUSSION THREAD

**Discussion by Cunya** [NVIDIA-Omniverse/PhysX#139] [Q&A] [answered]
I am trying to create triangle meshes with more than 130k triangles. At about 130k limit the objects seems to be getting corrupt, with random triangles here and there and the object getting squished along the x axis:
Original object:
![image](https://user-images.githubusercontent.com/1614691/236456708-a40fae83-0023-42ec-9475-114bdb35c8bf.png)
Triangle mesh (PxTriangleMesh) in PhysX Visual Debugger using 120k triangles of the data:
![image](https://user-images.githubusercontent.com/1614691/236457152-14249a6b-995a-4180-813c-c89393e9f136.png)
Triangle mesh (PxTriangleMesh) in PhysX Visual Debugger using 150k triangles of the data:
![image](https://user-images.githubusercontent.com/1614691/236463288-078085a3-a2f1-4ea1-bd5c-e4db2db1989a.png)
Triangle mesh (PxTriangleMesh) in PhysX Visual Debugger using all (430k triangles) of the data:
![image](https://user-images.githubusercontent.com/1614691/236463730-30b460a4-af38-4c12-950a-2402dde1e8cb.png)
Is there some limitation or setup I am not aware of or this is a bug? The same happens in PhysX v4.
Attaching the stl file with the geometry as well.
[High-triange-count.zip](https://github.com/NVIDIA-Omniverse/PhysX/files/11406437/High-triange-count.zip)

**Comment by PierreTerdiman** [comment, marked as answer]
Does it create issues in your app / game though? This could be just a limitation of PVD's renderer.

In PhysX itself there is an internal limit to the number of triangles supported by a BVH34 structure, but it should be well above these numbers. This is the only limit I am aware of, but as far as I remember it should only affect the internal BVH, not PVD's rendering.

**Comment by Cunya** [comment]
Yes, our testing since showed that it is probably only a bug/limitation in the visual debugger and the geometry is probably ok in PhysX itself. (I am only checking the overlap with the geometry and that seems to work ok.)
